### PR TITLE
Subscribers: link to Jetpack Cloud to manage subscribers instead of WP.com

### PR DESCRIPTION
**Pull request by simison**
Now that Jetpack Cloud is open to everyone, let's also link Jetpack sites to Jetpack Cloud, not just WP.com Atomic sites.

## Proposed changes:

* Remove link to WP.com subscriber management also for Jetpack sites

**Before**

Screenshot 2024-05-01 at 14 54 19

**After**

Screenshot 2024-05-01 at 14 53 07

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Enable subscriptions for the Jetpack site
* Check the link in the sidebar, and the link in the settings:

    Screenshot 2024-05-01 at 14 57 16

* Test also WP.com sites

### CODE REVIEW

**Comment by simison** [comment]
@DustyReagan (since I saw you created the redirect link in the JP link manager) — does this change work well for all different types of WP.com sites currently too? I know the change works well for self-hosted Jetpack sites.

File: projects/plugins/jetpack/_inc/client/newsletter/newsletter.jsx [hunk 1]
```diff
@@ -43,16 +41,12 @@ function Newsletter( props ) {
 			return '';
 		}
 
-		const source = isWpAdminInterface
```
Comment (simison) [inline projects/plugins/jetpack/_inc/client/newsletter/newsletter.jsx:46]: Alternatively, add a check here for WP.com sites.

File: projects/plugins/jetpack/modules/subscriptions.php [hunk 2]
```diff
@@ -1026,13 +1026,8 @@ public function add_subscribers_menu() {
 
 		$blog_id = Connection_Manager::get_site_id( true );
 
-		$source = 'jetpack-menu-calypso-subscribers';
-		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
```
Comment (simison) [inline projects/plugins/jetpack/modules/subscriptions.php:1030]: Alternatively, add a check here for WP.com sites.

**Review by DustyReagan — APPROVED** [review]
Looks good to me!

Just noting that the "Default" admin interface still has a "Subscribers" menu item under the "Users" menu, that continues to link to https://wordpress.com/subscribers/[site] This seems fine and expected to me since we've largely been leaving the "Default" admin interface unchanged.

**Comment by simison** [comment]
>Just noting that the "Default" admin interface still has a "Subscribers" menu item under the "Users" menu, that continues to link to [https://wordpress.com/subscribers/[site]](https://wordpress.com/subscribers/%5Bsite%5D) This seems fine and expected to me since we've largely been leaving the "Default" admin interface unchanged.

Yes, this is good for folks on "default" experience on WP.com simple & Atomic.

Thanks for confirming!

**Pull request merged**